File: vol1/java-examples/src/main/java/com/heatonresearch/aifh/distance/ChebyshevDistance.java
package com.heatonresearch.aifh.distance;

public class ChebyshevDistance extends AbstractDistance {

    @Override
    public double calculate(double[] position1, int pos1, double[] position2, int pos2, int length) {
        double result = 0;
        for (int i = 0; i < length; i++) {
            double d = Math.abs(position1[pos1 + i] - position2[pos2 + i]);
            result = Math.max(d, result);
        }
        return result;
    }
}

File: vol1/python-examples/lib/distance/chebyshev_distance.py
#!/usr/bin/env python
# Programmer: Chris Bunch (shatterednirvana@gmail.com)


import math


from abstract_distance import AbstractDistance


class ChebyshevDistance(AbstractDistance):


  def calculate_from_positions(self, position1, pos1, position2, pos2, length):
    result = 0.0
    for index in range(length):
      d = math.fabs(position1[pos1 + index] - position2[pos2 + index])
      result = max(d, result)
    return result
